File: IA.Web/src/app/dtos/serverSideDto.ts
import { ServerSideTotalsDto } from "./serverSideTotalsDto";

export class ServerSideDto<T>
{
    Data: T[];
    Count: number;
    Total: ServerSideTotalsDto;

    constructor()
    {
        this.Data = new Array();
        this.Count = 0;
        this.Total = new ServerSideTotalsDto();
    }
}

File: IA.DTOs/ServerSideDto.cs
using System.Collections.Generic;

namespace IA.DTOs
{
    public class ServerSideDto<T>
    {
        public List<T> Data { get; set; }
        public int Count { get; set; }

        public ServerSideDto()
        {
            Data = new List<T>();
            Count = 0;
        }
    }
}